src/test/mockFactory.ts

```diff
@@ -51,8 +59,42 @@ export class MockFactory {
       update: jest.fn(),
       validate: jest.fn(),
       custom: null,
-      provider: {} as any,
-    };
+      provider: MockFactory.createTestAzureServiceProvider(),
+      service: "serviceName",
+      artifact: "app.zip",
+    } as any as Service;
+  }
+
+  public static createTestFunctions(functionCount = 3) {
```

Comment:
We should add typings to these mock factory functions.

Reply:
Part of the reason for not having typings is my own not knowing what types they are 😆 

But agreed. I'll do some more digging here